packages/mobile/src/screens/edit-track-screen/fields/AccessAndSaleField/PremiumRadioField/PremiumRadioField.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect } from 'react'
+
+import { Dimensions, View } from 'react-native'
+
+import IconCart from 'app/assets/images/iconCart.svg'
+import { Text } from 'app/components/core'
+import { useSetTrackAvailabilityFields } from 'app/hooks/useSetTrackAvailabilityFields'
+import { makeStyles } from 'app/styles'
+import { useColor } from 'app/utils/theme'
+
+import type { TrackAvailabilitySelectionProps } from '../../../components/types'
+
+import { TrackPreviewField } from './TrackPreviewField'
+import { TrackPriceField } from './TrackPriceField'
+
+type PremiumRadioFieldProps = TrackAvailabilitySelectionProps
+
+const screenWidth = Dimensions.get('screen').width
+
+const messages = {
+  title: 'Premium (Pay to Unlock)',
+  description:
+    'Unlockable by purchase, these tracks are visible to everyone but only playable by users who have paid for access.'
+}
+
+const useStyles = makeStyles(({ spacing, palette }) => ({
+  root: {
+    width: screenWidth - spacing(22),
+    gap: spacing(4)
+  },
+  titleContainer: {
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  title: {
+    fontSize: 22,
+    marginTop: 0
+  },
+  selectedTitle: {
+    color: palette.secondary
+  },
+  disabledTitle: {
+    color: palette.neutralLight4
+  },
+  titleIcon: {
+    marginTop: 0,
+    marginRight: spacing(2.5)
+  },
+  subtitleContainer: {
+    marginTop: spacing(2)
+  },
+  subtitle: {
+    color: palette.neutral
+  }
+}))
+
+export const PremiumRadioField = (props: PremiumRadioFieldProps) => {
+  const { selected, disabled } = props
+  const { set: setTrackAvailabilityFields } = useSetTrackAvailabilityFields()
+  const styles = useStyles()
+
+  const secondary = useColor('secondary')
+  const neutral = useColor('neutral')
+  const neutralLight4 = useColor('neutralLight4')
+
+  const titleStyles: object[] = [styles.title]
+  if (selected) {
+    titleStyles.push(styles.selectedTitle)
+  } else if (disabled) {
+    titleStyles.push(styles.disabledTitle)
+  }
+
+  const titleIconColor = selected
+    ? secondary
+    : disabled
+    ? neutralLight4
+    : neutral
+
+  useEffect(() => {
+    if (selected) {
+      setTrackAvailabilityFields({
```

Comment:
mmm kinda nice to do it this way instead of in one big `handleChange` like we do in web

Reply:
i also dont like this, its just the pattern with all the other access-and-sale fields at this point. Im going to refactor to align with web. just wanted to do minimal pass here to at least lock in the basic feature

Reply:
Yeah I'm not sure which pattern I like better tbh. This is simpler to read but not centralized so it's hard to see all the logic at once

Reply:
yeah that's my feeling too, its weird that each child needs to handle this